pkg/mimir/mimir.go

```diff
@@ -170,10 +174,12 @@ func (c *Config) RegisterFlags(f *flag.FlagSet, logger log.Logger) {
 	f.StringVar(&c.NoAuthTenant, "auth.no-auth-tenant", "anonymous", "Tenant ID to use when multitenancy is disabled.")
 	f.BoolVar(&c.PrintConfig, "print.config", false, "Print the config and exit.")
 	f.DurationVar(&c.ShutdownDelay, "shutdown-delay", 0, "How long to wait between SIGTERM and shutdown. After receiving SIGTERM, Mimir will report not-ready status via /ready endpoint.")
+	f.DurationVar(&c.CostAttributionEvictionInterval, "cost-attribution-eviction-interval", 30*time.Minute, "Time interval at which inactive cost attributions will be evicted from the cache.")
+	f.DurationVar(&c.CostAttributionCoolDownDuration, "cost-attribution-cool-down-duration", 20*time.Minute, "Duration during which any cost attribution for a user will be marked as __overflow__ after exceeding the specified limit, prior to resetting the cache.")
```

Comment:
I'd suggest namespacing the cost-attribution flags with a dot
```suggestion
	f.DurationVar(&c.CostAttributionEvictionInterval, "cost-attribution.eviction-interval", 30*time.Minute, "Time interval at which inactive cost attributions will be evicted from the cache.")
	f.DurationVar(&c.CostAttributionCoolDownDuration, "cost-attribution.cool-down-duration", 20*time.Minute, "Duration during which any cost attribution for a user will be marked as __overflow__ after exceeding the specified limit, prior to resetting the cache.")
```

Reply:
addressed here https://github.com/grafana/mimir/pull/9733/commits/f4ebc07871ef79a0ed7ee31c48032a9fd42fe757#diff-d1a605160f2b763dbf8d1feea0506ac9ee143a51b6a303815cf56b411db44c4cR176
the cooldown duration is removed to be aligned with other DBs, also with cooldown together with eviction the logic becomes complicated.